pkgs/development/compilers/ghc/head.nix

```diff
@@ -15,16 +15,26 @@
 
 , # If enabled, GHC will be built with the GPL-free but slower integer-simple
   # library instead of the faster but GPLed integer-gmp library.
-  enableIntegerSimple ? false, gmp ? null
+  enableIntegerSimple ? targetPlatform != hostPlatform, gmp ? null, m4
 
 , # If enabled, use -fPIC when compiling static libs.
   enableRelocatedStaticLibs ? targetPlatform != hostPlatform
 
 , # Whether to build dynamic libs for the standard library (on the target
   # platform). Static libs are always built.
-  enableShared ? !targetPlatform.useAndroidPrebuilt
+  #
+  # We don't do this by default if target != host, or if we target windows.
+  enableShared ? (targetPlatform == hostPlatform && !targetPlatform.isWindows && !targetPlatform.useAndroidPrebuilt)
+
+, # Whetherto build terminfo.
+  enableTerminfo ? !targetPlatform.isWindows
 
 , version ? "8.5.20180118"
+, ghcRevision ? "e1d4140be4d2a1508015093b69e1ef53516e1eb6"
```

Comment:
Is it really necessary to parameterize all aspects of the `src` attribute as function arguments? I would assume that few people actually mess with these settings, and those who want to can easily override the `src` attribute as a whole.

Reply:
Right! that should better be handed like this once we have the generic expression. Will drop.